File: combine.rb
#!/usr/bin/env ruby

# Author:: Scott W Olesen (swo@mit.edu)

require 'arginine'

params = Arginine.parse do
  desc "combines two files columnwise. output has length of first file."
  opt :separator, :short => "F", :default => "\t"
  arg :fn1
  arg :fn2
end

open(params[:fn1]).each_line.zip(open(params[:fn2]).each_line) do |l1, l2|
  # if the second file has ended, just put in a blank instead
  l2 ||= ''
  puts "#{l1.chomp}#{params[:separator]}#{l2.chomp}"
end

File: combine.py
#!/usr/bin/env python

# author:: scott olesen (swo@mit.edu)

import argparse, fileinput

if __name__ == '__main__':
    p = argparse.ArgumentParser(description="combines two files columnwise. output has length of first file.", formatter_class=argparse.ArgumentDefaultsHelpFormatter)
    p.add_argument('--separator', '-F', default="\t")
    p.add_argument('input', nargs='+')
    args = p.parse_args()

    if len(args.input) < 2:
        raise RuntimeError('found only {} inputs, need at least 2'.format(len(args.input)))

    for lines in zip(*[fileinput.input(inp) for inp in args.input]):
        print(*[line.rstrip() for line in lines], sep="\t")
